Products/PleiadesEntity/browser/adapters/__init__.py

```diff
@@ -61,6 +61,67 @@ def memoize_all_methods(cls):
     return cls
 
 
+def revision_history(obj):
+    """Return revision history for an object.
+
+    Return value includes a `_sort_key` on each item in the returned data
+    so it can later be merged with output of workflow_history().
+    """
+    rt = getToolByName(obj, "portal_repository")
+    if rt is None or not rt.isVersionable(obj):
+        return []
+    try:
+        history = rt.getHistoryMetadata(obj)
+    except Unauthorized:
+        history = None
+    if not history:
+        return []
+    result = []
+    # Count backwards from most recent to least recent
+    for i in xrange(history.getLength(countPurged=False) - 1, -1, -1):
+        revision = history.retrieve(i, countPurged=False)
+        meta = revision['metadata']['sys_metadata']
+        userid = meta['principal']
+        modified = DateTime(meta['timestamp'], 'UTC')
+        modified._timezone_naive = True
+        result.append({
+            '_sort_key': modified,
+            'comment': meta['comment'],
+            'modified': modified.HTML4(),
+            'modifiedBy': userid,
+        })
+    return result
+
+
+def workflow_history(obj):
+    """Return workflow history for an object.
+
+    Return value includes a `_sort_key` on each item in the returned data
+    so it can later be merged with output of revision_history().
+    """
+    wf = getToolByName(obj, 'portal_workflow')
+    wf_name = wf.getDefaultChain()[0]
```

Comment:
I could iterate over the chain and get all events, but pretty sure it's not worth it.